tests/test_httpproxy_asyncio.py

```diff
@@ -170,15 +173,16 @@ def test_connect_failure(self):
         # Reset the proxy mock
         fake_proxy.reset_mock()
 
-        # pretend we got a sad response with a HTML error page
+        # For the sake of this test, pretend the credentials are incorrect so
+        # send a sad response with a HTML error page
         fake_proxy.pretend_to_receive(
             b"HTTP/1.0 401 Unauthorised\r\n\r\n<HTML>... some error here ...</HTML>"
         )
 
         # advance event loop because we have to let coroutines be executed
         self.loop.advance(1.0)
 
-        # *now* we should be complete
+        # *now* this future should have completed (
```

Comment:
I think the open paren is a typo?

Reply:
Yes, sadly I make a few of those random ones somehow. Suspect I had a different window in mind to type to but didn't focus it.